docs/design/README.md

```diff
@@ -0,0 +1,595 @@
+We use `cargo` to resolve dependencies between variants and packages, and we use custom build scripts to launch docker-hosted build environments.
+Packages are defined and built using RPM.
+
+These package builds take place in a docker environment using `docker build` commands and `buildx`.
+The toolchains used to compile Rust, C, C++ and Go software are distributed in a container image called the [bottlerocket-sdk].
+It is in the SDK environment that the package builds take place.
+
+Once all the software for a Bottlerocket image has been built into RPM images, a final image creation step occurs.
+A `docker build` command installs everything into a rooted image file.
+
+The toolchains necessary for each arch (x86_64 and aarch64) are available as separate SDK containers to allow for cross-compilation.
+The build host has the following requirements: sufficient versions of Docker, buildx, Cargo, Cargo Make and a few tools like lz4.
+The root of the Bottlerocket git repo is mounted and artifacts are created in `.cargo` and `build` directories.
+
+Certain custom tools for building, testing and publishing Bottlerocket ([buildsys], [pubsys], and [testsys]) are compiled from source during a Bottlerocket build.
+These are built using the host-installed Rust toolchain instead and not the SDK.
+Once these are built, they are used elsewhere in the build process.
+Buildsys, for example, is the tool that launches container build environments from `build.rs` scripts.
+
+[How the Bottlerocket build system works]: https://aws.amazon.com/blogs/opensource/how-the-bottlerocket-build-system-works/
+[cargo make]: https://github.com/sagiegurari/cargo-make
+[bottlerocket-sdk]: https://github.com/bottlerocket-os/bottlerocket-sdk
+[buildsys]: https://github.com/bottlerocket-os/bottlerocket/blob/3af909865c4da765f4afc787aaf5f50eea5c3989/tools/buildsys/Cargo.toml#L2
+[pubsys]: https://github.com/bottlerocket-os/bottlerocket/blob/3af909865c4da765f4afc787aaf5f50eea5c3989/tools/pubsys/Cargo.toml#L2
+[testsys]: https://github.com/bottlerocket-os/bottlerocket/blob/3af909865c4da765f4afc787aaf5f50eea5c3989/tools/testsys/Cargo.toml#L2
+
+### Enhancing the Build System
+
+The existing build system will remain intact with certain features and abstractions added to it.
+Twoliter will serve as a facade providing a new interface over the existing build system.
+Twoliter will understand how to start a Bottlerocket build using the OOTB user's sources, packages etc.
+It will inject these directories into the existing build system by setting existing and new `Makefile.toml` variables.
+
+Twoliter will eliminate the need for all build-host software other than `docker`.
+A bootstrapping container will be published that contains any software needed to kick off and sustain a `cargo make` build command.
+
+## Kits
+
+A central idea in this design is that the build system will be able to produce and consume Docker images that contain RPM packages.
+We call these images *kits*.
```

Comment:
Would a variant maintainer expected to be brining their own container image? For example, a container image which may contain precompiled binaries or build tools for adding binaries to the Bottlerocket rootfs.

Reply:
> Would a variant maintainer expected to be brining their own container image? For example, a container image which may contain precompiled binaries or build tools for adding binaries to the Bottlerocket rootfs.

Unless you want to publish your RPM packages for other Bottlerocket out-of-tree projects to consume, then you don't need to think or worry about "kits". If you aren't publishing RPMs to other teams or to the outside world, and if they are only used as part of your own image builds, then they are an implementation detail you don't need to worry about too much.